src/System.Net.Http/tests/FunctionalTests/ResponseStreamTest.cs

```diff
@@ -106,12 +106,31 @@ await client.GetAsync(Configuration.Http.RemoteEchoServer, HttpCompletionOption.
                 Task task = stream.ReadAsync(buffer, 0, buffer.Length, cts.Token);
                 cts.Cancel();
 
-                // Verify that the task completes successfully or is canceled.
+                // Verify that the task completed.
                 Assert.True(((IAsyncResult)task).AsyncWaitHandle.WaitOne(new TimeSpan(0, 5, 0)));
                 Assert.True(task.IsCompleted, "Task was not yet completed");
-                if (task.IsFaulted)
+
+                // Verify that the task completed successfully or is canceled.
+                if (PlatformDetection.IsWindows)
+                {
+                    // On Windows, we may fault because canceling the task destroys the request handle
+                    // which may randomly cause an ObjectDisposedException (or other exception).
+                    Assert.True(
+                        task.Status == TaskStatus.RanToCompletion ||
+                        task.Status == TaskStatus.Canceled ||
+                        task.Status == TaskStatus.Faulted);
```

Comment:
This assert will never fail; the only way it could fail is if task.IsCompleted was false, but we just verified it was true in an assert above.